aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -2344,37 +2344,81 @@ void fake_quantize_learnable_tensor_grad_kernel_cpu(
   });
 }
 
+template <typename SelfType>
+void _fake_quant_per_channel_cachemask_cpu_helper(
+    TensorIterator& iter,
+    TensorIterator& iter_mask,
+    const int64_t quant_min,
+    const int64_t quant_max) {
+
+  const auto& zero_point_dtype = iter.input_dtype(2);
+
+  if(at::isFloatingType(zero_point_dtype)){
+    // When zero_point is float, quantize mirroring affine quantizer equation
+    // Xq = Round(Xf * inv_scale + zero_point)
+    // where zero_point is in float.
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(zero_point_dtype, "fake_quantize_channel_cachemask_cpu_zero_point_handling", [&] {
+      // write mask
+      cpu_kernel(iter_mask, [=](SelfType self, float scale, scalar_t zero_point) -> bool {
+        float inv_scale = 1.0f / scale;
+        const auto qval = std::lrintf(zero_point + (self * inv_scale));
```

Comment:
why not use `std::nearbyintf()` to be consistent?

Reply:
This is mirroring numerics from `affine_quantizer_base.cpp`,  so keeping std::lrintf for consistency with the quantization implementation it is fakeQuant'ing.

```
/*
* Quantize value based on the following equation
* Xq = Round(Xf * inv_scale + zero_point)
* where zero_point is in float.
*
* Note: For the case of embedding quantization we will set zero_point
* to (-Xmin/scale), where Xmin is the min value in input tensor row.
*/
int quantize_val_float_qparams(float scale, float zero_point, float value, int qmin, int qmax) {
  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
  int qvalue;

  float inv_scale = scale == 0 ? 1.0f : 1.0f / scale;
  // NOLINTNEXTLINE(cppcoreguidelines-narrowing-conversions,bugprone-narrowing-conversions)
  qvalue = lrintf(value * inv_scale + zero_point);
  qvalue = std::max(qmin, std::min(qvalue, qmax));
  return qvalue;
}
```